xija/component/heat.py

```diff
@@ -400,13 +408,58 @@ def __init__(self, model, node,
         self.n_mvals = 1
         self.add_par('k', k, min=0.0, max=2.0)
 
+    def calc_earth_vis_from_grid(self, ephems, q_atts):
+        import astropy_healpix
+        healpix = astropy_healpix.HEALPix(nside=32, order='nested')
+
+        if not hasattr(self, 'earth_vis_grid'):
+            with fits.open(self.earth_vis_grid_path) as hdus:
+                hdu = hdus[0]
+                hdr = hdu.header
+                vis_grid = hdu.data / hdr['scale']  # 12288 x 100
+                self.__class__.earth_vis_grid = vis_grid
```

Comment:
Could you elaborate on why this is being set this way?

Reply:
Here and below, I just needed a place to cache the pre-computed data and grid points so they only get read in once.  Storing as class attributes seems convenient enough won't require reading the data again if you make a new model instance.  The other option I thought of was a module global, but this seemed more contained and localized (and I think this is more well-behaved for multi-processing on Windows, if we ever go that direction).  I also have used things like a global LazyDict that knows how to load data on demand, but that seemed like overkill.  Other ideas?

@mbaski @jskrist - in the MATLAB implementation it is important to recognize this one-time startup cost (about 1 sec in my profiling).  I assume ORviewer is using a persistent Python interpreter session for typical model evaluation?

Reply:
@taldcroft - The current implementation does use a persistent Python session.  The one time startup cost is good to know about, but it won't affect the MATLAB implementation.  When profiling the MATLAB code, I always take an average over a set of runs since the initial run can be affected by this kind of thing, and in practice, users will be running this model many times, so a small hit on the first run is acceptable.